### PR TITLE
ref: `data status` updates

### PR DESCRIPTION
**Pull request by jorgeorpinel**
Follow up to https://github.com/iterative/dvc.org/pull/3812#issuecomment-1237663329

### CODE REVIEW

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [inline content/docs/command-reference/data/status.md:53]: One question though: what's an `unknown` change? Does it apply to untracked files too? I didn't know where to mention that in the updated text.

**Comment by jorgeorpinel** [comment]
There may be other things for you to include based on pending comments under https://github.com/iterative/dvc.org/pull/3812#issuecomment-1237663329 @dberenbaum 

Thanks!

File: content/docs/command-reference/data/status.md
Comment (shcheklein) [inline content/docs/command-reference/data/status.md:53]: Again, let's not do a PR before we change this in DVC repo (and that is questionable). I mean DVC prefix. Please make a PR that is clearly fixing things first and easy to review

File: content/docs/command-reference/data/status.md
Comment (shcheklein) [inline content/docs/command-reference/data/status.md:86]: why removing it?

**Review by shcheklein — CHANGES_REQUESTED** [review]
@jorgeorpinel we go into another round of discussions here I feel, again there are copy edits + substnatial product changes together. It makes review complicated.

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [inline content/docs/command-reference/data/status.md:53]:
```suggestion
- `DVC committed changes` are new, modified, or deleted tracked files or directories
```

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [inline content/docs/command-reference/data/status.md:56]:
```suggestion
- `DVC uncommitted changes` are new, modified, or deleted tracked files or
```

**Review by jorgeorpinel — COMMENTED** [review]
Per https://github.com/iterative/dvc.org/pull/3924#pullrequestreview-1098210978

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:53]: I forgot to roll those back when quickly copy-pasting the important changes requested for this PR. Done now.

But changing things in docs before the core repo is something we do regularly: it's a circular conversation wrt terminology and user-facing text. Is that OK of should there be a heavier process around that?

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:86]: I suggest removing it because it's already used in the description.

> You asked me to make a PR with the most important changes from my feedback and IMO this is one of them: https://github.com/iterative/dvc.org/pull/3812#discussion_r963326344

Comment (shcheklein) [reply content/docs/command-reference/data/status.md:86]: https://github.com/iterative/dvc.org/pull/3812#discussion_r958632037 - the problem is it was discussed already. Question - do you have an actually strong opinion about this? :) (I'm personally fine and Dave was fine it seems)

File: content/docs/command-reference/data/status.md
Comment (shcheklein) [inline content/docs/command-reference/data/status.md:48]: sections names were taken from https://dvc.org/doc/command-reference/status#status for consistency (italic style), again doesn't feel worth creating a PR

could you please highlight any substantial changes in this text please? 

I don't know how to make the process better, but reading 4 paragraphs bc of a few words is just to painful :(

File: content/docs/command-reference/data/status.md
Comment (shcheklein) [inline content/docs/command-reference/data/status.md:67]: `directories tracked as a whole` - I prefer the previous one. "directory tracked as whole" is too complicated, tbh we should not have other directories also

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md]: I didn't realize it had been discussed (I didn't read all the resolved comments). I still think it's unnecessary since the `--granular` example can still refer to the one in the description.

It's not a auper strong opinion but it does go against good practices for cmd ref examples, I think: they should add special value to the doc, not just cover obvious cases. And to avoid redundancy in general.

But anyway, I rolled it back in 725c7e4f3 since it was discussed by them.

File: content/docs/command-reference/data/status.md
Comment (shcheklein) [inline content/docs/command-reference/data/status.md:48]: Q: are there changes besides extra spaces?

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md]: On process: There's a reason I reviewed that PR and left feedback (gathering best practices). I didn't intend to make a PR, much less for you to have to review it. Maybe a better process would be to communicate directly to fill any info. gaps (why is this effort happening in the first place) and then decide whether to stop something before getting to this point, or let it run its course.

<details>

<summary>On the style (minor details)</summary>

I used back ticks because this is under a code block where the same strings as shown in monotype, so this made the doc look consistent internally. Our formats and styles should be able to evolve I think, and this was easier to read IMO:

image

> Compare to [now](https://dvc.org/doc/command-reference/data/status). 

Also, the format here is already different to https://dvc.org/doc/command-reference/status#local-workspace-status because the latter has sub-lists that we didn't include here (but could).

</details>

I agree that part is minor, but even after I rolled it back (f3fa7c6f9) [the GH diff](https://github.com/iterative/dvc.org/pull/3924/files#diff-fa3d177a2271cc250d345c45de9a7fd83d793ad890ccc4fa7abef1b7a7960921L46-R65) looks almost the same, sorry (the whole block replaced).

> please highlight any substantial changes in this text?

- Improves clarity of explanations in general (and fixes syntax and typos in some cases).
- Mentions scenarios (when each state happens or what to do with it).
- Links to relevant commands.

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:48]: Yes. Please see https://github.com/iterative/dvc.org/pull/3924#discussion_r964412125.

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md]: You're right. I just thought it wouldn't hurt to be emphatic. Changed back to "tracked directories" in 1cf235bac.

**Review by shcheklein — APPROVED** [review]

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:53]: One question pending here cc @dberenbaum @skshetry 🙏🏼 

> what's an unknown change?

Comment (dberenbaum) [reply content/docs/command-reference/data/status.md:53]: For example, let's say you clone a repo with a tracked directory but haven't done `dvc fetch/pull`. You have an md5 in the .dvc file but the directory is empty. If you modify the directory (for example, add a file to it), DVC can tell that the directory doesn't match the md5 but has no info about the granular changes to the files in that directory.

File: content/docs/command-reference/data/status.md
Comment (skshetry) [inline content/docs/command-reference/data/status.md:48]: Let's not introduce the term `file records`, it's not the same thing as "file hashes are recorded". The `status` says "file hash(es)".

From `dvc status` docs:
> not in cache: An output exists in the workspace, and the corresponding file hash in the [dvc.lock](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#dvclock-file) or [.dvc](https://dvc.org/doc/user-guide/project-structure/dvc-files) file is up to date, but there is no corresponding cache file or directory.

File: content/docs/command-reference/data/status.md
Comment (skshetry) [inline content/docs/command-reference/data/status.md:53]: Let's call it `added` to keep it in sync with the output.
```suggestion
- _DVC committed changes_ are added, modified, or deleted tracked files or
```

Comment (skshetry) [reply content/docs/command-reference/data/status.md:53]: Same on _DVC uncommitted changes_.

File: content/docs/command-reference/data/status.md
Comment (skshetry) [inline content/docs/command-reference/data/status.md:93]: Whitespace is significant.

File: content/docs/command-reference/data/status.md
Comment (skshetry) [inline content/docs/command-reference/data/status.md:97]: Whitespace is significant.

File: content/docs/command-reference/data/status.md
Comment (skshetry) [inline content/docs/command-reference/data/status.md:3]: The command name is `data`. I am not sure what we get by replacing it.

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:53]: > DVC can tell that the directory doesn't match the md5 but has no info about the granular changes to the files in that directory

So it only applies to `--granular` right? (Assuming that generalizing that example is the one scenario for this.) I added a note in that option instead of trying to cover it up here (it still counts as "new, modified, or deleted").

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:48]: Changed to "file hashes".

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:53]: I'm not using back ticks because this doesn't intend to reflect the exact output. I think "added" is misleading and the output itself seems to be up for debate. Please see/comment in https://github.com/iterative/dvc/pull/7943#pullrequestreview-1097028252.

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:93]: Not sure how this was deleted. Maybe linter or just a typo. Restoring whitespaces!

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:3]: We are specific about what we mean by data. Docs should explain 🙂

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [inline content/docs/command-reference/data/status.md:78]: > I don't understand this comment tbh.
_From https://github.com/iterative/dvc.org/commit/aca9d8a1a170c6e818d93f26f545f93f866c14de#r83797370_

Do you mean you're not familiar with the underlying situation this refers to @shcheklein? Please see https://github.com/iterative/dvc.org/pull/3924#discussion_r966209543 if so. It's kind of confusing (maybe I didn't even get it right, waiting for another review from the core team).

Another alternative is to not even document this specifically, if it's evident what `unknown` means in the CLI output (self-documenting).

Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:78]: Another alternative is to not even document this specifically, if it's evident what unknown means in the CLI output (self-documenting).

```suggestion
  directories.
```

Comment (shcheklein) [reply content/docs/command-reference/data/status.md:78]: I might understand this if read the comment (i haven't yet), but I would expect to be able to understand it from the text, right?

do you mean that some `files (not granular changes) can be reported as "unknown" if (when? `as DVC tracks` is not informative)`?

Comment (jorgeorpinel) [reply content/docs/command-reference/data/status.md:78]: Yeah it's not super easy to explain briefly. How about this?
```suggestion
  directories. Note that granular changes may have `unknown` state if tracked
  [directory contents](doc/user-guide/project-structure/internal-files#directories)
  are not present in the local cache (see also `dvc fetch`).
```

Comment (shcheklein) [reply content/docs/command-reference/data/status.md:78]: Status for files inside DVC-tracked directories may be reported as "unknown" if information about directories is missing from the cache yet (see ...)

File: content/docs/command-reference/data/status.md
Comment (dberenbaum) [reply content/docs/command-reference/data/status.md:53]: @skshetry Do you know if it can ever apply to non-granular data?

Comment (skshetry) [reply content/docs/command-reference/data/status.md:53]: yeah, it only applies to granular.

**Review by dberenbaum — APPROVED** [review]

File: content/docs/command-reference/data/status.md
Comment (dberenbaum) [reply content/docs/command-reference/data/status.md:53]: @jorgeorpinel It's a good point, but @skshetry also has a reasonable rationale for using "added," so what do you think about continuing discussion in another issue and leaving "added" to get this merged?

**Comment by dberenbaum** [comment]
@jorgeorpinel What's the status of this PR? I see some unresolved discussions, but it's also approved, so up to you at this point I think.

File: content/docs/command-reference/data/status.md
Comment (jorgeorpinel) [inline content/docs/command-reference/data/status.md:10]: This

File: content/docs/command-reference/status.md
Comment (jorgeorpinel) [inline content/docs/command-reference/status.md:9]: and this

**Review by jorgeorpinel — COMMENTED** [review]
OK well I guess it's all approved now. The only change after that is improved cross-tip admon/links:

**Pull request merged**
